code/modules/mob/living/basic/vermin/mouse.dm

```diff
@@ -118,6 +118,10 @@
 	// Now if we were't ACTUALLY gibbed, spawn the dead mouse
 	if(!gibbed)
 		var/obj/item/food/deadmouse/mouse = new(loc)
+		if(istype(mouse.loc.loc, /obj/machinery/microwave))//If our mouse is dying in place holder in a microwave, put the dead mouse in the microwave
```

Comment:
This is pretty ugly, maybe instead of this there's some general solution we can find for the mouse dying inside a mob holder?

Reply:
Think I've got something that looks a bit better now

Reply:
Does putting a dead mouse inside a mob holder even work?
I think probably what you want to do is destroy the mob holder if the mouse dies and then try to place the dead mouse in the same place the mouse was if possible, which would handily also prevent you from needing a check specifically for microwaves.

Reply:
I'm not sure if putting the dead mouse item into the placeholder as it dies can work, but that's not what I'm doing
Currently I'm placing the dead mouse in the location the holder is in, and the holder cleans itself up since it no longer has the mouse in it when the mouse mob qdel's itself at the end of its death.
This works perfectly fine with containers like the chef's hat, backpacks, etc., or if the holder has been deep fried and not picked up and the mouse inside dies
Maybe there's some really obscure way already existing that a mouse could die in a placeholder on a player that isn't in a container that already exists on live that this wouldn't address that I don't know about that would lead to a dead mouse being in a person's contents but not accessible since it died in their pocket or something weird

But adding the dead mouse to the microwave's ingredients list is necessary in addition to its contents because otherwise it will not release the dead mouse when the microwave finishes its cook timer, leading to any additional rat bodies being accessible only by disassembling it

Reply:
Rather than put the check for adding a mouse to the ingredients list in the mouse typepath it might be preferable to do it in `Entered` in the microwave, to pick up any future case of an item suddenly entering the microwave's contents while it is actively cooking.

Reply:
Been rather occupied irl, but I've opted for preventing the dead mouse food object being added in addition to the mouse mob being gibbed since the gibbing creates mouse meat and the dead body only ended up in a burnt mess when the microwave then promptly finishes cooking anyway